client/cli/src/lib.rs

```diff
@@ -625,16 +613,16 @@ where
 			}
 		});
 
-	if config.rpc_http.is_none() {
+	if config.rpc_http.is_none() || cli.rpc_port.is_some() {
```

Comment:
Why? Isn't the port setting the `rpc_http`?

Reply:
On `Configuration` yes but here it's on `RunCmd` (cli). CLI is the user input and if it is provided it must override the one configured (if there is one)